src/go/rpk/pkg/config/redpanda_yaml.go

```diff
@@ -174,14 +174,14 @@ type (
 	}
 
 	RpkKafkaAPI struct {
-		Brokers []string `yaml:"brokers,omitempty" json:"brokers"`
-		TLS     *TLS     `yaml:"tls,omitempty" json:"tls"`
+		Brokers []string `yaml:"brokers,omitempty" json:"brokers,omitempty"`
+		TLS     *TLS     `yaml:"tls,omitempty" json:"tls,omitempty"`
```

Comment:
omitempty is for the nullness, rather than emitting `"tls": null`, this will just leave out the field.
More importantly, I want to ensure that the yaml renders the same(ish) as json. The formatType function errors if the json and yaml tags are not the same -- so these need to be equal.